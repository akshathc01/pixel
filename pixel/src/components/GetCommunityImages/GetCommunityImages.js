import React, { useEffect, useState } from 'react';
import ShowCommunityImages from '../ShowCommunityImages/ShowCommunityImages';

const GetCommunityImages = () => {
    const [images, setEvents] = useState([]);
    const [search, setSearch] = useState('')

    useEffect(() => {
        fetch(`http://localhost:3002/communityImages`)
            .then(res => res.json())
            .then(data => setEvents(data))
    }, [])

    return (
        <section data-testid="test-getCommunityImages" className="services-container">
            <div className="text-center">
                <div className="form-group row d-flex justify-content-center pt-5">
                    <input type="" className="form-control border border-danger col-5 " placeholder="Type for finding images" onChange={e => { setSearch(e.target.value) }} />
                </div>
            </div>
            <div className="d-flex justify-content-center align-items-center">
                <div className="w-100 row pb-5  justify-content-center">
                    {
                        images.filter((val) => {
                            if (search == "") {
                                return val
                            } else if (val.title.toLowerCase().includes(search.toLocaleLowerCase()))
                                return val
                        }).map(image => <ShowCommunityImages image={image}></ShowCommunityImages>)
                    }
                </div>
            </div>

        </section>
    );
};

export default GetCommunityImages;