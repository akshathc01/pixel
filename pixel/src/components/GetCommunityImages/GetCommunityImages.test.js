import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import GetCommunityImages from "./GetCommunityImages"


afterEach(() => {
    cleanup()
})

test('test', () => {
    expect(true).toBe(true)
})

test('it will render GetCommunityImages component', () => {
    render(<GetCommunityImages />)
    const getCommunityImagesElement = screen.getByTestId('test-getCommunityImages')
    expect(getCommunityImagesElement).toBeInTheDocument()
})


test('matches snapshot for GetCommunityImages', () => {
    const tree = renderer.create(<GetCommunityImages />).toJSON()
    expect(tree).toMatchSnapshot()
})