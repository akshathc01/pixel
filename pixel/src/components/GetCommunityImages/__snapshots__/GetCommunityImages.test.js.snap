// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for GetCommunityImages 1`] = `
<section
  className="services-container"
  data-testid="test-getCommunityImages"
>
  <div
    className="text-center"
  >
    <div
      className="form-group row d-flex justify-content-center pt-5"
    >
      <input
        className="form-control border border-danger col-5 "
        onChange={[Function]}
        placeholder="Type for finding images"
        type=""
      />
    </div>
  </div>
  <div
    className="d-flex justify-content-center align-items-center"
  >
    <div
      className="w-100 row pb-5  justify-content-center"
    />
  </div>
</section>
`;
